backup_integrator_operator/src/charm.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""backup-integrator charm."""
+
+import logging
+import pathlib
+import typing
+
+import ops
+from charms.backup_integrator.v0.backup import BackupDynamicRequirer
+
+logger = logging.getLogger(__name__)
+
+
+class BackupIntegratorCharm(ops.CharmBase):
+    """Charm the service."""
+
+    _CHARM_OPT_DIR = pathlib.Path("/opt/backup-integrator-charm")
+
+    def __init__(self, *args: typing.Any):
+        """Construct.
+
+        Args:
+            args: Arguments passed to the CharmBase parent constructor.
+        """
+        super().__init__(*args)
+        self._requirer = BackupDynamicRequirer(charm=self)
+        self.framework.observe(self.on.backup_relation_created, self._reconcile)
+        self.framework.observe(self.on.backup_relation_changed, self._reconcile)
+        self.framework.observe(self.on.config_changed, self._reconcile)
+        self.framework.observe(self.on.upgrade_charm, self._reconcile)
```

Comment:
I'm not sure about this, but shouldn't we listen to the leader change events ? I've the feeling that changing the leader may have an impact on the content of the backup